### ISSUE TITLE
`Darling\PHPJsonUtilities\classes\decoders\JsonDecoder->decode()` does not properly decode objects within an array

### ISSUE THREAD

**Issue by sevidmusic** [sevidmusic/PHPJsonUtilities#35]
`Darling\PHPJsonUtilities\classes\decoders\JsonDecoder->decode()` does not properly decode objects within an array. 

For example:
```
<?php

require_once(
    __DIR__ .
    DIRECTORY_SEPARATOR .
    'vendor' .
    DIRECTORY_SEPARATOR .
    'autoload.php'
);

use \Darling\PHPJsonUtilities\classes\decoders\JsonDecoder;
use \Darling\PHPJsonUtilities\classes\encoded\data\Json;
use \Darling\PHPTextTypes\classes\strings\Id;

$jsonDecoder = new JsonDecoder();

$id = new Id();

$array = [$id];

$jsonEncodedArray = new Json($array);

$decodedArray = $jsonDecoder->decode($jsonEncodedArray);

var_dump($decodedArray);

// actual output:
/**
array(1) {
  [0] =>
  class stdClass#7 (0) {
  }
}
 */

// expected output:
/**
array(1) {
  [0] =>
  class Darling\PHPTextTypes\classes\strings\Id#2 (2) {
    private string $string =>
    string(77) "Pf5o0ZOe9K8bCXWj8nGOxohGGhyyHeCEmbuvVXH6wUD9mLInjy1zi5gqwjUMnqz2dkKobMe8zT5ba"
    private Darling\PHPTextTypes\interfaces\strings\Text $text =>
    class Darling\PHPTextTypes\classes\strings\AlphanumericText#4 (2) {
      private string $string =>
      string(77) "Pf5o0ZOe9K8bCXWj8nGOxohGGhyyHeCEmbuvVXH6wUD9mLInjy1zi5gqwjUMnqz2dkKobMe8zT5ba"
      private Darling\PHPTextTypes\interfaces\strings\Text $text =>
      class Darling\PHPTextTypes\classes\strings\Text#5 (1) {
        ...
      }
    }
  }
}
 */


```

**Comment by sevidmusic** [comment]
JsonDecoder still does not decode objects in arrays, instead they are left encoded as `Json`. The should be decoded along with all the other values in the array.

For example, the following output from `JsonDecoderExample.php` shows that objects that existed in the original array are decoded to json strings:

```
array(10) {
  [0] =>
  int(1)
  [1] =>
  bool(true)
  [2] =>
  bool(false)
  [3] =>
  NULL
  [4] =>
  string(6) "string"
  [5] =>
  array(0) {
  }
  [6] =>
  string(93) "{"__class__":"Darling\\\\PHPTextTypes\\\\classes\\\\strings\\\\Text","__data__":{"string":"adbfecg"}}"
  'baz' =>
  array(1) {
    'secondary_id' =>
    string(574) "{"__class__":"Darling\\\\PHPTextTypes\\\\classes\\\\strings\\\\Id","__data__":{"text":"{\\"__class__\\":\\"Darling\\\\\\\\PHPTextTypes\\\\\\\\classes\\\\\\\\strings\\\\\\\\AlphanumericText\\",\\"__data__\\":{\\"text\\":\\"{\\\\\\"__class__\\\\\\":\\\\\\"Darling\\\\\\\\\\\\\\\\PHPTextTypes\\\\\\\\\\\\\\\\classes\\\\\\\\\\\\\\\\strings\\\\\\\\\\\\\\\\Text\\\\\\",\\\\\\"__data__\\\\\\":{\\\\\\"string\\\\\\":\\\\\\"ZHCuZgogh5WEUCSChHr3OsIrx0UgfdtmWd68dXHXkw1WkNyMj98jePr7EHDaP1gBi5zwbV\\\\\\"}}\\",\\"string\\":\"...
  }
  'foo' =>
  string(3) "bar"
  'id' =>
  string(544) "{"__class__":"Darling\\\\PHPTextTypes\\\\classes\\\\strings\\\\Id","__data__":{"text":"{\\"__class__\\":\\"Darling\\\\\\\\PHPTextTypes\\\\\\\\classes\\\\\\\\strings\\\\\\\\AlphanumericText\\",\\"__data__\\":{\\"text\\":\\"{\\\\\\"__class__\\\\\\":\\\\\\"Darling\\\\\\\\\\\\\\\\PHPTextTypes\\\\\\\\\\\\\\\\classes\\\\\\\\\\\\\\\\strings\\\\\\\\\\\\\\\\Text\\\\\\",\\\\\\"__data__\\\\\\":{\\\\\\"string\\\\\\":\\\\\\"1SJ8Ce9hHcDmXW67PrBAVcYDetwhM7MJjbPGnmou4SP9TmOYZ0vVTJEroDk3\\\\\\"}}\\",\\"string\\":\\"1SJ8Ce9h"...
}
```

**Comment by sevidmusic** [comment]
resolved in 70ef83d
